File: backend/src/main/java/com/mycompany/backend/service/UsersService.java
package com.mycompany.backend.service;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.mycompany.backend.dao.UsersDao;
import com.mycompany.backend.dto.Pager;
import com.mycompany.backend.dto.User;

@Service
public class UsersService {
   @Autowired
   private UsersDao usersDao;

   public int getCount() {
      return usersDao.count();
   }
   
   public int getCountSearch(String keyword) {
	      return usersDao.countSearch(keyword);
   }
   
   public String getAuthority(String user_id) {
	   return usersDao.userAuthority(user_id);
   }
   
   public List<User> getList(Pager pager) {
      return usersDao.selectByPage(pager);
   }
   
   public List<User> getSearchList(Pager pager) {
	      return usersDao.selectBySearch(pager);
	   }
   
   public User getUser(String user_id) {
	  return usersDao.selectById(user_id);
   }

   public int delete(String user_id) {
      return usersDao.deleteById(user_id);
   }
   
   public Integer totalPay(String user_id) {
	   return usersDao.totalPay(user_id);
   }
}

File: frontend/services/usersService.js
angular.module("app")
  .factory("usersService", function($http) {
    //변수 선언
    const BASE_URL = "http://localhost:8080/user";
    //서비스 객체 리턴
    return {
      list: function(pageNo=1,keyword=null) {
        const promise = $http.get(BASE_URL, {params:{pageNo,keyword}});
        console.log("pageNo:"+pageNo);
        console.log("promise:"+promise);
        return promise;
      },
      read: function(user_id) {
        const promise = $http.get(BASE_URL + "/" + user_id);
        return promise;
      },
      delete: function(user_id){
        const promise=$http.delete(BASE_URL + "/" + user_id);
        return promise;
      }
    }
  });